pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -556,6 +556,34 @@ func (p *Pod) groupTotalCount() (int, error) {
 	return gtc, nil
 }
 
+// podGroupIndex returns the value of GroupIndexLabel for the pod being reconciled at the moment.
+func (p *Pod) podGroupIndex() (*int, error) {
+	if podGroupName(p.pod) == "" {
+		return nil, fmt.Errorf("pod doesn't have a '%s' label", GroupNameLabel)
+	}
+	podGroupTotalCount, err := p.groupTotalCount()
+	if err != nil {
+		return nil, fmt.Errorf("pod group total count is invalid: %v", err)
+	}
+	groupIndex, ok := p.Object().GetAnnotations()[GroupIndexLabel]
```

Comment:
Why reading from annotations actually, not labels?

Reply:
Good catch, fixed. Test is in progress

Reply:
Done